app/controllers/Application.java

```diff
@@ -2021,4 +2143,67 @@ private static String reasonForDisablingTuning(TuningJobDefinition tuningJobDefi
     }
     return paramValueMap;
   }
+
+  private static Map<String, String> authenticateUser(String username, String password, String url) throws Exception {
+    logger.info("Making authentication call to " + url);
+    Map<String, String> postCallEntityParams = new HashMap();
+    postCallEntityParams.put(USERNAME, username);
+    postCallEntityParams.put(PASSWORD, password);
+    HttpResponse response = httpPostCall(url, postCallEntityParams);
+    HttpEntity entity = response.getEntity();
+    Map<String, String> responseMap = new HashMap();
+    if (entity != null) {
+      String responseEntity = EntityUtils.toString(entity);
+      JSONObject result = new JSONObject(responseEntity);
+      if (result.has(ERROR_KEY)) {
+        responseMap.put(ERROR_KEY, result.get(ERROR_KEY).toString());
+      } else {
+        logger.info(result.get(STATUS));
+        if (result.get(STATUS) != null && result.get(STATUS).toString().equals(SUCCESS)) {
+          responseMap.put(STATUS, result.get(STATUS).toString());
+          responseMap.put(SESSION_ID_KEY, result.get(AZKABAN_SESSION_ID_KEY).toString());
+        }
+      }
+    } else {
+      logger.error("Empty response entity encountered while authenticating " + username);
+      responseMap.put(ERROR_KEY, "Something went wrong while processing the request");
+    }
+    return responseMap;
+  }
+
+  private static HttpResponse httpGetCall(String url, Map<String,String> queryParams)
+      throws IOException, URISyntaxException {
+    if (url == null) {
+      throw new IllegalArgumentException("URL cannot be NULL for GET call");
+    }
+    URIBuilder builder = new URIBuilder(url);
+    logger.info("Making a GET call to URL " + url);
+    if (queryParams != null) {
+      for (Map.Entry<String, String> entry : queryParams.entrySet()) {
+        builder.setParameter(entry.getKey(), entry.getValue());
+      }
+    }
+    URI getUri = builder.build();
+    HttpGet request = new HttpGet(getUri);
+    CloseableHttpClient client = HttpClients.createDefault();
+    return client.execute(request);
+  }
+
+  private static HttpResponse httpPostCall(String url, Map<String,String> postParams)
+      throws IOException {
```

Comment:
IOException is not thrown from this method

Reply:
CloseableHttpClient.execute throws IOException that's why this method is throwing it.